cpp/oneapi/dal/backend/primitives/lapack/potrf_dpc.cpp

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/detail/profiler.hpp"
+
+#include "oneapi/dal/backend/primitives/lapack/solve.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+namespace detail {
+
+struct potrf_params {
+    std::int64_t n;
+    std::int64_t lda;
+    mkl::uplo uplo;
+};
+
+template <mkl::uplo uplo, typename Float, ndorder layout>
+auto get_potrf_params(const ndview<Float, 2, layout>& x) {
+    potrf_params result;
+
+    ONEDAL_ASSERT(layout == ndorder::c);
+
+    result.lda = x.get_stride(0);
+    result.n = x.get_dimension(1);
+    result.uplo = ident_uplo(uplo);
+
+    return result;
+}
+
+template <mkl::uplo uplo, typename Float, ndorder layout>
+std::int64_t potrf_scratchpad_size(sycl::queue& queue, const ndview<Float, 2, layout>& x) {
+    ONEDAL_ASSERT(x.has_data());
+    const auto [ncount, nlda, nuplo] = get_potrf_params<uplo>(x);
+    return mkl::lapack::potrf_scratchpad_size<Float>(queue, nuplo, ncount, nlda);
+}
+
+template <mkl::uplo uplo, typename Float, ndorder layout>
+sycl::event potrf_factorization(sycl::queue& queue,
+                                ndview<Float, 2, layout>& x,
+                                array<Float>& scratchpad,
+                                const event_vector& deps) {
+    ONEDAL_PROFILER_TASK(potrf_kernel, queue);
+
+    ONEDAL_ASSERT(x.has_mutable_data());
+    ONEDAL_ASSERT(scratchpad.has_mutable_data());
+    const auto [ncount, nlda, nuplo] = get_potrf_params<uplo>(x);
+
+#ifdef ONEDAL_ENABLE_ASSERT
+    const auto scratchpad_real_count = scratchpad.get_count();
+    const auto scratchpad_want_count = potrf_scratchpad_size<uplo>(queue, x);
+    ONEDAL_ASSERT(scratchpad_real_count >= scratchpad_want_count);
+#endif
```

Comment:
Is `#ifdef ONEDAL_ENABLE_ASSERT` really required here to avoid overhead in no assert mode? If no, previous comment is applicable

Reply:
Changed to use of ```[[maybe_unused]]`` instead